model/LTIDeliveryTool.php

```diff
@@ -108,4 +114,13 @@ public function getLinkedDeliveryExecutions(core_kernel_classes_Resource $delive
 	    }
 	    return $returnValue;
 	}
+
+    /**
+     * @return AssignmentService
+     */
+    protected function getAssignmentService()
+    {
+        $registry = AssignmentServiceRegistry::getRegistry();
```

Comment:
Registry is probably not the best approach here, a simple configurable service would do